kiota_serialization_multipart/multipart_serialization_writer.py

```diff
@@ -0,0 +1,277 @@
+        """
+        raise NotImplementedError()
+
+    def write_bytes_value(self, key: Optional[str], value: bytes) -> None:
+        """Writes the specified byte array as a base64 string to the stream with an optional
+        given key.
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (bytes): The byte array to be written.
+        """
+        if value and len(value) > 0:
+            self._stream.write(value)
+
+    def write_collection_of_primitive_values(
+        self, key: Optional[str], values: Optional[List[T]]
+    ) -> None:
+        """Writes the specified collection of primitive values to the stream with an optional
+        given key.
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            values (Optional[List[T]]): The collection of primitive values to be written.
+        """
+        raise NotImplementedError()
+
+    def write_collection_of_enum_values(
+        self, key: Optional[str], values: Optional[List[Enum]]
+    ) -> None:
+        """Writes the specified collection of enum values to the stream with an optional given key.
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            values Optional[List[Enum]): The enum values to be written.
+        """
+        raise NotImplementedError()
+
+    def write_enum_value(self, key: Optional[str], value: Optional[Enum]) -> None:
+        """Writes the specified enum value to the stream with an optional given key.
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Optional[Enum]): The enum value to be written.
+        """
+        raise NotImplementedError()
+
+    def write_collection_of_object_values(
+        self, key: Optional[str], values: Optional[List[U]]
+    ) -> None:
+        """Writes the specified collection of model objects to the stream with an optional
+        given key.
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            values (Optional[List[U]]): The collection of model objects to be written.
+        """
+        raise NotImplementedError()
+
+    def write_object_value(
+        self, key: Optional[str], value: Optional[U], *additional_values_to_merge: U
+    ) -> None:
+        """Writes the specified model object to the stream with an optional given key.
+        Args:
+            key (Optional[str]): The key to be used for the written value. May be null.
+            value (Parsable): The model object to be written.
+            additional_values_to_merge (tuple[Parsable]): The additional values to merge to the
+            main value when serializing an intersection wrapper.
+        """
+        temp_writer = self._create_new_writer()
+
+        if isinstance(value, MultipartBody):
```

Comment:
nit: I'd negate the condition and throw early to avoid the arrow head code style